### DISCUSSION TITLE
Kargo Notifications

### DISCUSSION THREAD

**Discussion by soudaburger** [akuity/kargo#2624] [Q&A] [answered]
When using just `ArgoCD`, we were getting notifications into slack with the application/cluster/user information for sync events.  Now that we have `Kargo` doing some of that as a prototype, the `kargo-controller` is technically initiating the sync via `ArgoCD`.  Is there a native mechanism for notifications within Kargo? Or is there a way to get the username of the invoker passed to `ArgoCD` ?

**Comment by krancour** [comment, marked as answer]
See #1305

Notifications have been on our radar for some time. We are currently days away from the v0.9.0 release, which executes a major pivot in how promotion processes are defined. This is the first phase of a three phase plan to make promotion processes pluggable. What we anticipate is developing notification capabilities as plugins to that system.

**Comment by soudaburger** [comment]
Would this potentially include something like Github PR notifications to show a diff to developers making changes?  Or is that something I should ask for in a separate discussion topic?

**Comment by krancour** [comment]
> Would this potentially include something like Github PR notifications

There is this open issue: https://github.com/akuity/kargo/issues/821

That has more to do with showing diffs _in the UI._

But in terms of incorporating something like that into a notification, it's certainly within the realm of possibility to do something like that, but the current plan really has more to do with building the _capability_ to incorporate reusable, custom promotion logic and less to do with actually implementing such plug-ins on a first-party basis.

I expect community-drive/third-party efforts to create useful plugins, including ones for notifications. tbh, I also expect that Kargo maintainers may implement some plugins ourselves, but will probably offer them as premium features of Akuity Platform.

**Reply by soudaburger** [reply]
Appreciate the responses!! Thank you!
